File: openbis/source/java/ch/systemsx/cisd/openbis/generic/shared/basic/dto/EntityKind.java
/*
 * Copyright 2008 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.systemsx.cisd.openbis.generic.shared.basic.dto;

import ch.systemsx.cisd.openbis.generic.shared.basic.ISerializable;

/**
 * The <i>GWT</i> counterpart to
 * {@link ch.systemsx.cisd.openbis.generic.shared.dto.properties.EntityKind}.
 * 
 * @author Christian Ribeaud
 */
public enum EntityKind implements ISerializable
{
    MATERIAL("Material"), EXPERIMENT("Experiment"), SAMPLE("Sample"), DATA_SET("Data Set");

    private final String description;

    private EntityKind(final String description)
    {
        this.description = description;
    }

    public final String getDescription()
    {
        return description;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/entitytype/EntityKind.js
/**
 * @author pkupczyk
 */
define([ "stjs", "as/dto/common/Enum" ], function(stjs, Enum) {
	var EntityKind = function() {
		Enum.call(this, [ "MATERIAL", "EXPERIMENT", "SAMPLE", "DATA_SET" ]);
	};
	stjs.extend(EntityKind, Enum, [ Enum ], function(constructor, prototype) {
	}, {});
	return new EntityKind();
})